src/App.js

```diff
@@ -1,16 +1,86 @@
 import { CssBaseline } from '@mui/material';
 import { ThemeProvider } from '@mui/material/styles';
-import React from 'react';
+import React, { useEffect, useCallback } from 'react';
 import Div100vh from 'react-div-100vh';
 import './App.css';
 import { initAnalytics, logPageView } from './analytics';
 import Head from './components/Head/Head';
 import MapPage from './components/MapPage/MapPage';
 import theme from './theme';
+import { useDispatch } from 'react-redux';
+import { userDataSlice } from 'reducers/user';
+
+/**
+ * Handle checking geolocation on an interval.
+ *
+ * @param {NodeJS.Interval} interval Interval to clear on callback run.
+ * @param {(status: boolean) => void} setter Setter to run when the callbacks run.
+ */
+function geolocationInterval(interval, setter) {
+  navigator.geolocation.getCurrentPosition(
+    () => {
+      setter(true);
+      clearInterval(interval);
+    },
+    () => {
+      setter(false);
+      clearInterval(interval);
+    }
+  );
+}
+
+/**
+ * Check geolocation perms and set an onchange handler.
+ * @async
+ * @param {(status: boolean) => void} setter Setter to run based on initial query.
+ */
+async function queryGeolocationPerms(setter) {
+  const perms = await navigator.permissions.query({ name: 'geolocation' });
+  setter(perms.state === 'granted');
+  perms.addEventListener('change', () => {
+    setter(perms.state === 'granted');
+  });
+}
 
 function App(props) {
   initAnalytics();
   logPageView();
+  const dispatch = useDispatch();
```

Comment:
since this is running on the app top level, maybe it's worth refactoring it into a hook like `useUserLocationStatus`